### DISCUSSION TITLE
Cypress 10. Plugin events - cannot add more than one listener to events

### DISCUSSION THREAD

**Discussion by kpturner** [cypress-io/cypress#23465] [Plugins] [answered]
I have found since using Cypress 10 that if I have multiple listeners on, for example `before:browser:launch` then only the last listener added fires.  For example, if I do this

```
on('before:browser:launch', (browser, launchOptions) => {
    console.log('FIRST');
});
on('before:browser:launch', (browser, launchOptions) => {
    console.log(SECOND');
});
on('before:browser:launch', (browser, launchOptions) => {
    console.log('THIRD');
});
```
Then only `THIRD` gets logged.

This has meant that various plugins that have functions listening to `before:browser:launch` have simply stopped working as expected. The only one that works is the last one to add a listener to the event.

Is this a bug or by design?

**Comment by lmiller1990** [comment, marked as answer]
This sounds like a bug to me - we should be merging all the listeners together, so multiple plugins can use the same hook. 

I wonder if it's getting clobbered here: https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/data/EventRegistrar.ts#L40

I think we should open an issue, sounds like a bug.

**Reply by kpturner** [reply]
Please see https://github.com/cypress-io/cypress/issues/23496
